src/OmniSharp.MSBuild/ProjectLoadListener.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Composition;
+using System.IO;
+using Microsoft.Build.Execution;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Eventing;
+using OmniSharp.MSBuild.Notification;
+
+namespace OmniSharp.MSBuild
+{
+    [Export(typeof(IMSBuildEventSink))]
+    public class ProjectLoadListener : IMSBuildEventSink
+    {
+        internal const string TargetFrameworkPropertyName = "TargetFramework";
+        internal const string TargetFrameworkVersionPropertyName = "TargetFrameworkVersion";
+        private readonly ILogger _logger;
+        private readonly IEventEmitter _eventEmitter;
+        private const string MSBuildProjectFullPathPropertyName = "MSBuildProjectFullPath";
+
+        [ImportingConstructor]
+        public ProjectLoadListener(ILoggerFactory loggerFactory, IEventEmitter eventEmitter)
+        {
+            _logger = loggerFactory.CreateLogger<ProjectLoadListener>();
+            _eventEmitter = eventEmitter;
+        }
+
+        public void ProjectLoaded(ProjectLoadedEventArgs args)
+        {
+            try
+            {
+                var projectFilePath = args.ProjectInstance.GetPropertyValue(MSBuildProjectFullPathPropertyName);
+                if (string.IsNullOrEmpty(projectFilePath))
+                {
+                    // This should never be true but we're being extra careful.
+                    return;
+                }
+
+                var targetFramework = GetTargetFramework(args.ProjectInstance);
+                if (string.IsNullOrEmpty(targetFramework))
+                {
+                    // This should never be true but we're being extra careful.
+                    return;
+                }
+
+                if(args.References == null)
+                {
+                    return;
+                }
+
+                _eventEmitter.ProjectInformation(projectFilePath, targetFramework, args.References);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Unexpected exception got thrown from project load listener: " + ex);
+            }
+        }
+
+        // Internal for testing
+        internal static string GetTargetFramework(ProjectInstance projectInstance)
+        {
+            var targetFramework = projectInstance.GetPropertyValue(TargetFrameworkPropertyName);
```

Comment:
should we look at `<TargetFrameworks>` too?

Reply:
Done. Thanks!